aten/src/ATen/native/sparse/SparseTensorMath.cpp

```diff
@@ -153,7 +153,8 @@ SparseTensor pow_sparse_scalar(const SparseTensor& t, Scalar value) {
 SparseTensor& div_out_sparse_zerodim(SparseTensor& r, const SparseTensor& t, const Tensor& value);
 
 Tensor div_sparse(const Tensor& self, const Tensor& value) {
-  Tensor result = at::empty({0}, self.options());
+  auto promotedType = at::result_type(self, value);
```

Comment:
it doesn't seem like a great idea to call this promotedType when there is a function called promote_types that can return something different than result_type.

Reply:
hmm... result_type does call promote_types though, just that it does so while considering "category." I'm hesitant to call the type resultType because in other contexts it can be cast back into the type of a specified result object, and it seems more readable to be consistent across these functions. This is the intermediate highest-precision promoted type where operations are performed.

E.g. if I do float_tensor *= double_tensor the promoted type is double and operations are performed on doubles, but the result will still be float. The name of the result_type function itself comes from numpy.

Reply:
changed to `commonDtype` since that's equivalent to what we have in TensorIterator.cpp. Does that work for you?